### ISSUE TITLE
Implement contact list for Matrix contacts like Roster in XMPP Jabber

### ISSUE THREAD

**Issue by MurzNN** [vector-im/riot-web#4488]
At now Riot track only Rooms in left sidebar. Even in People category there are also rooms, but with 2 persons, this is not direct contact to user.

So there are no way for users to manage classic contact list, like Roster in XMPP Jabber, User list in ICQ, Android contacts list, etc.

Without contact list it's too hard to find specific user contact when inviting it to room, or when you want to start new direct chat - I must remember it's email or Matrix ID for find him. 

Also very often I rename users in contact list to my name, for easily finding. For example, user can fill in name only nickname "Fox" that not so unique, and I rename it to "Alexey Fox Ivanov (friend of Sergey Petrov)" to remember who is it.

So will be good to implement Matrix private user contact list (something like Jabber roster), for manage private Matrix contacts list (not rooms) in separate Riot tab, and shows items from contact list in "Start a chat" and "Invite new room members" windows.

**Comment by MTRNord** [comment]
A direct chat is always a "normal group chat" in matrix and I think there were multiple Issues about it that got reject because direct chats are equal to rooms. But petnames for rooms are planned I think ( https://github.com/vector-im/riot-web/issues/3130 ).

**Comment by MurzNN** [comment]
Yes, the main problem is direct chats is group chats and "People" is only custom tag for multi-user room. So I can rename it, add users, remove users and after this - I lost contacts. We need static contact list, independent to rooms.

**Comment by MurzNN** [comment]
Seems that "Contact list" API or something other to manage contact list is missed in [current Matrix spec](http://matrix.org/docs/spec/client_server/r0.2.0.html), So we must implement this at first on Matrix Server side?

**Comment by MTRNord** [comment]
Well I thik first wait for votes of the community :)

**Comment by turt2live** [comment]
Related: https://github.com/vector-im/riot-web/issues/2930

**Comment by uhoreg** [comment]
Also related: #1646

**Comment by cuibonobo** [comment]
Strong vote for this. I've been looking for a replacement to our aging XMPP server and Matrix+Riot *almost* fits the bill, but people on a team are used to being able to see the whole roster list of who they are able to talk to in an organization. Unfortunately Riot doesn't provide this, so user onboarding is harder than it needs to be.

Basically, I'm echoing @ara4n's [comment on 2930](https://github.com/vector-im/riot-web/issues/2930#issuecomment-285991090).

**Comment by DiThi** [comment]
This needs to be done as an additional level of abstraction over rooms:

- I should be able to add someone to my contact list, either from a button in their profile or from a matrix ID.
- When clicking a contact in the contact list (or from a button in the profile), it should _instantly_ show an "empty room", and not to try to create an actual room until I send a message.
- Invites should clearly show it's not a "room" but a direct 1:1 conversation.
- Add an option to automatically accept invites from direct messages. Maybe with some additional conditions (e.g. only from people seen in channels).
- Direct message rooms should hide a lot of room features: renaming, moderating... People tries to rename direct message rooms assuming it's only visible to them, like when you rename a contact in a contact list.

Every single time my team tries Riot we find the private message UX awkward, confusing and slow, and not worth switching yet. Even something like showing common rooms in the user profile would help a lot, to avoid making duplicate private conversations again and again.

Using https://riot.im/develop/

**Comment by UnaSalisVictis** [comment]
I signed up just to say, yes, this is needed, badly, i get it requires code on both server and client side but, this is what keeps 1/2 of my friends from using riot.im, and insisting we should use wire instead since. it has proper contact lists. 

i love the phone camera app and alot of aspects of riot, i would like some more theme options, and the settings to be....more user friendly(the layout on pc is as if your using a phone, and not one person i have on riot with me has anything but complaints about that aspect of the stupid thing..

they could just implement the ability to make 1:1 and non 1:1 rooms that are "locked"/"forever"  for those who tick that option, this would probably be easier then adding a treditional contact list. 

this is really what keeps a huge number of people  i know from using it....

**Comment by banholzer** [comment]
I can confirm. We're just migrating from an xmpp environment to matrix / riot and the users are confused, because they don't have a roster.

**Comment by avirdoz** [comment]
Upvote for this feature.

**Comment by DinizCabreira** [comment]
I'll definitely upvote for this too.

Riot.im / Matrix is a dream come true in the downwards spiral that IM applications have become. I sincerely believe that if we want to drag people into it we need a way for them to be able to see their «friends» and contact them easily, in the way they are used to with other applications.

**Comment by MurzNN** [comment]
Maybe this can be implemented via private groups (communities), to which users will can add contacts without sharing this info to added user and others? 
So users will create private community "Personal contacts" and manage it via add/remove all contacts that wanted.

**Comment by MurzNN** [comment]
And maybe more easier implementation way: now we can add personal `ignore` flag to each matrix user. After this - in Riot we can view list of ignored users in `Settings` window.

So we can implement via same way another flag "contactlist", and mark needed users via it.
And in Matrix client implement page "Contacts" with list of matrix users, marked with this flag.

**Comment by DinizCabreira** [comment]
From my completely ignorant viewpoint, this last solution seems like it
would work!

--
Denís Fernández Cabrera
Gatonegro desenho
http://gatonegro.me

2018-03-07 9:30 GMT+01:00 Alexey Murz Korepov <notifications@github.com>:

**Comment by menturion** [comment]
+1 for this feature request.

Plus "contactlist" (friend) flagging through rest API.

**Comment by bergentroll** [comment]
At least it would be very convenient to have user-specific aliases for rooms/direct chats.

**Comment by syhe** [comment]
@bergentroll That feature is tracked in #3130

**Comment by bergentroll** [comment]
@syhe, oh, indeed, thank you!

**Comment by neg3ntropy** [comment]
On the mobile end of the spectrum, it would say that it's expected nowadays that the address book be managed in a way similar to whatsapp/telegram, that is lookup users by phone number and allow contacts to see that I can be reached.
Privacy-wise, some granularity of control is welcome, but leverage of address books is the key to adoption.

**Comment by hekmon** [comment]
I can't believe matrix is out of beta without this feature 😲
+1 much needed

**Comment by miloshavlicek** [comment]
Hi, any new info about this feature?

**Comment by adamrpostjr** [comment]
Checking in on this issue, any news?

**Comment by Mikaela** [comment]
Is the upstream spec issue https://github.com/matrix-org/matrix-doc/issues/556 or is this expected to be resolved by https://github.com/matrix-org/matrix-doc/pull/2199 ?
